ddtrace/contrib/internal/openai/utils.py

```diff
@@ -247,14 +255,23 @@ def _is_async_generator(resp):
 
 
 def _loop_handler(span, chunk, streamed_chunks):
-    """Sets the openai model tag and appends the chunk to the correct index in the streamed_chunks list.
-
-    When handling a streamed chat/completion response, this function is called for each chunk in the streamed response.
     """
+    Sets the openai model tag and appends the chunk to the correct index in the streamed_chunks list.
+    When handling a streamed chat/completion/responses,
+    this function is called for each chunk in the streamed response.
+    """
+
     if span.get_tag("openai.response.model") is None:
-        span.set_tag("openai.response.model", chunk.model)
-    for choice in chunk.choices:
-        streamed_chunks[choice.index].append(choice)
+        if hasattr(chunk, "type") and chunk.type.startswith("response."):
+            response = getattr(chunk, "response", None)
+            model = getattr(response, "model", "")
+        else:
+            model = getattr(chunk, "model", "")
+        span.set_tag_str("openai.response.model", model)
+    # Only run if the chunk is a completion/chat completion
+    if hasattr(chunk, "object") and "completion" in chunk.object.lower() and hasattr(chunk, "choices"):
+        for choice in chunk.choices:
+            streamed_chunks[choice.index].append(choice)
```

Comment:
```suggestion
        for choice in getattr(chunk, "choices", []):
            streamed_chunks[choice.index].append(choice)
```
Isn't this sufficient here? Since we only really care about iterating if chunk.choices is an actual list of objects